### PR TITLE
Normalize URL without a scheme, resolves #271

### LINKED ISSUE #271: Normalize URL without a scheme
I mean that if someone attempts to normalize "google.com" should our normalizer return null (like it does today) or should it should we return http://google.com

Maybe in the aggresive normalizer ?

### PR DESCRIPTION
**Pull request by sebastian-nagel**
- prefix URL without a scheme with `http://`

### CODE REVIEW

File: src/main/java/crawlercommons/filters/basic/BasicURLNormalizer.java [hunk 1]
```diff
@@ -143,6 +149,15 @@ public String filter(String urlString) {
...
+            if (!hasSchemePattern.matcher(urlString).find()) {
+                // no protocol/scheme : try to prefix http://
+                try {
+                    url = new URL("http://" + urlString);
+                } catch (MalformedURLException e1) {
+                }
+            }
+        }
+        if (url == null) {
             LOG.info("Malformed URL {}", urlString);
```
Comment (kkrugler) [inline src/main/java/crawlercommons/filters/basic/BasicURLNormalizer.java:161]: Not related to your code change, but logging at info level here seems wrong - I would change to debug

**Review by kkrugler — APPROVED** [review]
lgtm - one comment about logging level (for unchanged code)

**Review by Chaiavi — APPROVED** [review]
Perfect, thank you S

File: src/main/java/crawlercommons/filters/basic/BasicURLNormalizer.java (hunk 1 above)
Comment (sebastian-nagel) [reply src/main/java/crawlercommons/filters/basic/BasicURLNormalizer.java:161]: Ok, done.

**Pull request merged**